lib/valueflow.cpp

```diff
@@ -2197,6 +2197,8 @@ static bool bifurcate(const Token* tok, const std::set<nonneg int>& varids, cons
         const Token* start = var->declEndToken();
         if (!start)
             return false;
+        if (start->strAt(-1) == ")" || start->strAt(-1) == "}")
+            return false;
```

Comment:
Actually, just adding `depth` to the recursive call solves the endless recursion, but I think this is more correct. An improvement would be to actually handle the initialization instead.